sqlalchemy_exasol/pyodbc.py

```diff
@@ -164,5 +167,139 @@ def is_disconnect(self, e, connection, cursor):
+
+        quoted_view_name_string = self.quote_string_value(tables[0][2])
+        quoted_view_schema_string = self.quote_string_value(tables[0][1])
+        sql_statement = (
+            "/*snapshot execution*/ SELECT view_text "
+            f"FROM sys.exa_all_views WHERE view_name = {quoted_view_name_string} "
+            f"AND view_schema = {quoted_view_schema_string}"
+        )
+        result = connection.execute(sql.text(sql_statement)).scalar()
+        return result if result else None
+
+    @reflection.cache
+    def get_table_names(self, connection, schema, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super().get_table_names(connection, schema, **kw)
+        tables = self._tables_for_schema(connection, schema, table_type="TABLE")
+        return [self.normalize_name(row.table_name) for row in tables]
+
+    @reflection.cache
+    def get_view_names(self, connection, schema=None, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super().get_view_names(connection, schema, **kw)
+        tables = self._tables_for_schema(connection, schema, table_type="VIEW")
+        return [self.normalize_name(row.table_name) for row in tables]
+
+    def has_table(self, connection, table_name, schema=None, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super().has_table(connection, table_name, schema, **kw)
+        tables = self.get_table_names(
+            connection=connection,
+            schema=schema,
+            table_name=table_name,
+            **kw
+        )
+        return self.normalize_name(table_name) in tables
+
+    def _get_schema_names_query(self, connection, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super()._get_schema_names_query(connection, **kw)
+        return "/*snapshot execution*/ " + super()._get_schema_names_query(connection, **kw)
+
+    @reflection.cache
+    def _get_columns(self, connection, table_name, schema=None, **kw):
+        if self._is_sql_fallback_requested(**kw):
+            return super()._get_columns(connection, table_name, schema, **kw)
+
+        tables = self._tables_for_schema(connection, schema=schema, table_name=table_name)
+        if len(tables) != 1:
+            return []
+
+        # get_columns_sql originally returned all columns of all tables if table_name is None,
+        # we follow this behavior here for compatibility. However, the documentation for Dialects
+        # does not mention this behavior:
+        # https://docs.sqlalchemy.org/en/13/core/internals.html#sqlalchemy.engine.interfaces.Dialect
+        quoted_schema_string = self.quote_string_value(tables[0].table_schem)
+        quoted_table_string = self.quote_string_value(tables[0].table_name)
+        sql_statement = "/*snapshot execution*/ {query}".format(query=self.get_column_sql_query_str())
+        sql_statement = sql_statement.format(schema=quoted_schema_string, table=quoted_table_string)
```

Comment:
We need to check if we can replace this query with a odbc metadata function.

Reply:
See https://github.com/exasol/sqlalchemy-exasol/issues/139